### DISCUSSION TITLE
Ability to copy chat in lazer.

### DISCUSSION THREAD

**Discussion by s1iqbal** [ppy/osu#16635] [Ideas]
This should include a way to copy beatmap links that you can /np.

**Comment by peppy** [comment]
Can you explain the scenario in which you'd use it with `/np`? Wouldn't you usually be clicking the links?

**Reply by Rekunan** [reply]
one i can think of is sending beatmap links to ppl in like discord w/o needing to dl or go through web/browser

**Reply by s1iqbal** [reply]
It decreases the amount of steps needed when needing to share beat-maps.

If friend A gives me a beat map, I can quickly copy that link and share it on different platforms or to another friend in osu lazer rather than follow a 3 step process each time. This is especially true if I am in a middle of a song and during a break or something, I need to quickly copy or paste a link. Even more prevalent in multi where I might not want to alt tab since host can start 'whenever'.

Also, not everyone has amazing computers. Most people might not have their browser open or might not want to open their browser.

I believe it is more or a QOL feature rather than it being a necessary feature.

**Reply by peppy** [reply]
So if you could copy the links from the beatmap overlay, this would solve your issue? Or being able to right click on a link and choose to copy it?

To rephrase, this is about being able to copy links specifically, not *all text* in chat?

**Reply by s1iqbal** [reply]
Yes, it is about copying just the hyperlink, not the whole text for the feature of copying /np. 

This would be a lot faster than following the flow of opening the beatmap overlay through details and then copying the link. Because at that point, you might as well just open the link to a new browser.

Less steps, the better.
